text/0000-deprecate-fromstr.md

```diff
@@ -0,0 +1,81 @@
+- Feature Name: deprecate_fromstr
+- Start Date: 2020-05-11
+- RFC PR: TBD
+- Rust Issue: TBD
+
+# Summary
+[summary]: #summary
+
+Deprecate [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) in favor of [`TryFrom<&str>`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html)
+
+# Motivation
+[motivation]: #motivation
+
+`FromStr` was created when `TryFrom` did not exist, `FromStr` is now superfluous.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Basic implementation:
+```rust
+use std::convert::TryFrom;
+use std::num::ParseIntError;
+
+#[derive(Debug, PartialEq)]
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+impl From<&str> for Point {
+    type Error = ParseIntError;
+
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        let coords: Vec<&str> = s
+            .trim_matches(|p| p == '(' || p == ')')
+            .split(',')
+            .collect();
+
+        let x_fromstr = i32::try_from(coords[0])?;
+        let y_fromstr = i32::try_from(coords[1])?;
+
+        Ok(Point {
+            x: x_fromstr,
+            y: y_fromstr,
+        })
+    }
+}
+```
+
+Example:
+```rust
+let p = Point::try_from("(1,2)");
+assert_eq!(p.unwrap(), Point { x: 1, y: 2 })
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+- Replace all `FromStr` implementation in `std` with the corresponding `TryFrom<&str>`
+- Rewrite `str::parse()` to use `TryFrom<&str>`
+- Mark `FromStr` as deprecated
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Backward compatibility will be broken
```

Comment:
This... isn't a drawback; as far as I understand it it's a non-starter.

Reply:
Ok, I couldn't find anything on the policy around backward compatibility, I'll close the PR.

Reply:
> I couldn't find anything on the policy around backward compatibility

What places did you look? It could be good to have something more explicitly stated there for the future.

One reason it might not be easy to find is that stability is kind of a "core tenet" of the language and standard library and just permeates everything. One place I found was https://blog.rust-lang.org/2014/10/30/Stability.html

Reply:
I think the canonical source is still the API Evolution RFC: https://rust-lang.github.io/rfcs/1105-api-evolution.html

Reply:
Thanks for the link.
I searched for "rust deprecation policy" and "rust backward compatibility" with DuckDuckGo which led me nowhere. However a search on Google got me plenty of results but not the link you sent.

This should probably be in the Readme.